schema/event/createEventStep3.schema.ts

```diff
@@ -0,0 +1,36 @@
+import { number, object } from "yup"
+
+export const CreateEventStep3Schema = object().shape(
+  {
+    general: number().when("general", (val) => {
+      if (val === 0) {
+        return number().notRequired().min(1, "Require at least 1 ticket")
+      } else {
+        return number().notRequired()
+      }
+    }),
+    vip: number()
+      .notRequired()
+      .when("vip", (val) => {
+        if (val === 0) {
+          return number().notRequired().min(1, "Require at least 1 ticket")
+        } else {
+          return number().notRequired()
+        }
+      }),
+    reservedSeats: number()
+      .nullable()
+      .when("reservedSeats", (val) => {
+        if (val === 0) {
+          return number().notRequired().min(1, "Require at least 1 ticket")
+        } else {
+          return number().notRequired()
+        }
+      })
+  },
+  [
+    ["general", "general"],
+    ["vip", "vip"],
+    ["reservedSeats", "reservedSeats"]
+  ]
```

Comment:
Just wondering... what is this for?

Reply:
This is for fixing the circular dependency issue. In Yup validation schema, if I need to check the field's value in its own checking functions, then it will result in a circular dependency. this solves that (according to StackOverflow I saw, which works lol)